Makefile.rules

```diff
@@ -1090,7 +1090,24 @@ testlibgap: ${LIBGAPTESTS}
 		$(v) -A -l $(top_srcdir) -q -T --nointeract >$(v).out && \
 		diff $(top_srcdir)/$(v).expect $(v).out && ) :
 
-.PHONY: testlibgap
+KERNELTESTS := $(addprefix tst/testkernel/,dstruct)
+
+# run a test in tst/testkernel
+tst/testkernel/%: obj/tst/testkernel/%.lo libgap.la
+	$(QUIET_LINK)$(LINK) $(GAP_LDFLAGS) $^ $(GAP_LIBS) -o $@
+
+clean: clean-testkernel
+clean-testkernel:
+	rm -f ${KERNELTESTS}
+
+# run all the tests in tst/testkernel
+testkernel: ${KERNELTESTS}
+	$(foreach v,$^, \
+		echo "Running $(v) ..." && \
+		$(v) -A -l $(top_srcdir) -q -T --nointeract >$(v).out && \
+		diff $(top_srcdir)/$(v).expect $(v).out && ) :
+
+.PHONY: testlibgap testkernel
```

Comment:
Nice that you added `testkernel`, but this also should be hooked up to the CI, else those tests still never are run by anybody. A minimal patch for that cold look like this:
```patch
diff --git a/.travis.yml b/.travis.yml
index b11834b54..f04b91c7f 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -136,7 +136,7 @@ matrix:
           - libreadline-dev:i386
 
     # test error reporting and compiling as well as libgap
-    - env: TEST_SUITES="testspecial test-compile testlibgap"
+    - env: TEST_SUITES="testspecial test-compile testlibgap testkernel"
 
     # test Julia integration
     - env: TEST_SUITES="testinstall" JULIA=yes CONFIGFLAGS="--disable-Werror"
diff --git a/etc/ci.sh b/etc/ci.sh
index 009b241ce..b75d4c52c 100644
--- a/etc/ci.sh
+++ b/etc/ci.sh
@@ -167,6 +167,10 @@ GAPInput
     make testlibgap
     ;;
 
+  testkernel)
+    make testlibgap
+    ;;
+
   *)
     if [[ ! -f  $SRCDIR/tst/${TEST_SUITE}.g ]]
     then
```